### DISCUSSION TITLE
Build a functionnal librealsense2 with its cuda support

### DISCUSSION THREAD

**Discussion by aurelien-enchanted-tools** [OE4T/meta-tegra#1330] [Q&A]
Hello all, 

yocto version: langdale
meta-tegra version: langdale
meta-intel-realsense version: langdale
Jetson board: Xavier Nx

To add cuda support of the librealsense2 for the Intel RealSense D435I, I writed a recipes-support/librealsense/librealsense2_2.53.1.bbappend:
```
PACKAGECONFIG += "\
    cuda \
    rsusb \
    "

inherit cuda
PACKAGECONFIG[cuda] = "-DBUILD_WITH_CUDA:BOOL=ON, -DBUILD_WITH_CUDA:BOOL=OFF, ${CUDA_DEPENDS}"
```
It builds. But with my application bases on opencv, build with the sdk, which runs with the librealsense, I obtain after several minutes these following errors and an abort:
```
[…]
NVMAP_IOC_GET_FD failed: Bad address
NVMAP_IOC_GET_FD failed: Bad address
NVMAP_IOC_GET_FD failed: Bad address
NVMAP_IOC_GET_FD failed: Bad address
PosixMemGetSize:122 FD from Handle failed : Bad address
NVMAP_IOC_GET_FD failed: Bad address
[…]
```
Dmesg logs:
```
[…]
[ 2247.143376] nvmap: __nvmap_dmabuf_fd() NvMap: FD limit is crossed for uid 1000
[ 2247.176545] nvmap: __nvmap_dmabuf_fd() NvMap: FD limit is crossed for uid 1000
[ 2247.410582] nvmap: __nvmap_dmabuf_fd() NvMap: FD limit is crossed for uid 1000
[ 2247.443816] nvmap: __nvmap_dmabuf_fd() NvMap: FD limit is crossed for uid 1000
[…]
```
I don’t know where it comes from, why, and how to investigate it. Any ideas ? Maybe something missing ? Thank you in advance.

**Comment by kekiefer** [comment]
Is your application responsible for allocating buffers? There is a file descriptor leak somewhere, and the NV memory allocator is complaining that there are too many. You can check `ulimit` to see if the limit is unreasonably low, but likely you'll have to track down the leak.
